generation/internal/node.go

```diff
@@ -13,30 +13,80 @@ func npmTaskDefined(ls labels.LabelSet, task string) bool {
 	return ls[labels.DepsNode].Tasks[task] != ""
 }
 
-func nodeTestSteps(ls labels.LabelSet, packageManager string) []config.Step {
+func nodePackageManager(ls labels.LabelSet) string {
+	if ls[labels.PackageManagerYarn].Valid {
+		return "yarn"
+	}
+	return "npm"
+}
+
+func nodeRunCommand(ls labels.LabelSet, task string) string {
+	if task == "test" {
+		return fmt.Sprintf("%s test", nodePackageManager(ls))
+	}
+	return fmt.Sprintf("%s run %s", nodePackageManager(ls), task)
+}
+
+func nodeInitialSteps(ls labels.LabelSet) []config.Step {
+	steps := []config.Step{
+		checkoutStep(ls[labels.DepsNode]),
+	}
+
+	if ls[labels.DepsNode].HasLockFile {
+		steps = append(steps, config.Step{
+			Type:    config.OrbCommand,
+			Command: "node/install-packages",
+			Parameters: config.OrbCommandParameters{
+				"pkg-manager": nodePackageManager(ls),
+			}})
+	} else {
+		steps = append(steps, config.Step{
+			Comment: "Update the default install command as the project doesn't have a lock file",
+			Type:    config.OrbCommand,
+			Command: "node/install-packages",
+			Parameters: config.OrbCommandParameters{
+				"cache-path":          "~/project/node_modules",
+				"override-ci-command": fmt.Sprintf("%s install", nodePackageManager(ls)),
+			}})
+	}
```

Comment:
~ A potential simplification, we don't mind losing the comment

```suggestion
        installParams := config.OrbCommandParameters{
	        "pkg-manager": nodePackageManager(ls),
	}}
	if !ls[labels.DepsNode].HasLockFile {
		installParams = config.OrbCommandParameters{
			"cache-path":          "~/project/node_modules",
			"override-ci-command": fmt.Sprintf("%s install", nodePackageManager(ls)),
	}
	steps = append(steps, config.Step{
			Type:    config.OrbCommand,
			Command: "node/install-packages",
			Parameters: installParams})
	}
```